instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig/xrayconfig.go

```diff
@@ -26,9 +25,10 @@ import (
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 )
 
+// TODO: Currently does nothing but should support pulling the propagated
+// trace context from attributes depending on the type of Lambda event
 func xrayEventToCarrier([]byte) propagation.TextMapCarrier {
-	xrayTraceID := os.Getenv("_X_AMZN_TRACE_ID")
-	return propagation.HeaderCarrier{"X-Amzn-Trace-Id": []string{xrayTraceID}}
+	return propagation.HeaderCarrier{}
```

Comment:
I don't think this change is necessary.  This package provides a set of helpers for configuring the Lambda instrumentation with opinionated defaults, but is not required to use the instrumentation.  Leaving it as-is can minimize the impact to users who already have a working setup with these defaults while still having the expected linking behavior.

Reply:
But having this become the parent was part of the confusion people were seeing that led to changing it to a link.

Reply:
I support removing the `_X_AMZN_TRACE_ID` environment variable reference here because it doesn't conform to the updated spec.

Eventually this code should be updated to actually provide a carrier from the input, but I don't think that should preclude removing the environment variable usage.

Reply:
This is not the Lambda instrumentation and does not need to conform to any specification.  It is a helper provided to make it easier for users to have an experience that aligns with that of using X-Ray.

Reply:
It is lambda specific. If it continued to extract the environment variable then there would have to be a new `eventToCarrier` created to handle lambda events and ignore the environment variable and this one would be unused?

Reply:
@Aneurysm9 can this be resolved?

Reply:
I recommend that we not remove this functionality, but should remove it from the `WithRecommendedOptions`.  

There are two reasons I see for this change to the PR:
1. This allows a path for current users to keep current functionality.
2. There does seem to be a missing propagation here.  I'm not fluent enough if X-Ray to know what should be in the X-Amzn-Trace-Id header (would the local trace-id and span id work?) but it would seem without it all requests made would be orphaned.